src/full/Agda/TypeChecking/MetaVars.hs

```diff
@@ -1135,29 +1146,26 @@ checkSubtypeIsEqual a b = do
   ((a, b), equal) <- SynEq.checkSyntacticEquality a b
   unless equal $ do
     cumulativity <- optCumulativity <$> pragmaOptions
-    reduce (unEl b) >>= \case
-      Sort sb -> reduce (unEl a) >>= \case
+    abortIfBlocked (unEl b) >>= \case
+      Sort sb -> abortIfBlocked (unEl a) >>= \case
         Sort sa | cumulativity -> equalSort sa sb
-                | otherwise    -> return ()
-        MetaV{} -> patternViolation
+                             | otherwise    -> return ()
         Dummy{} -> return () -- TODO: this shouldn't happen but
                              -- currently does because of generalized
                              -- metas being created in a dummy context
-        _ -> patternViolation
-      Pi b1 b2 -> reduce (unEl a) >>= \case
+        a -> patternViolation (unblockOnAnyMetaIn a) -- TODO: can this happen?
+      Pi b1 b2 -> abortIfBlocked (unEl a) >>= \case
         Pi a1 a2
-          | getRelevance a1 /= getRelevance b1 -> patternViolation
-          | getQuantity  a1 /= getQuantity  b1 -> patternViolation
-          | getCohesion  a1 /= getCohesion  b1 -> patternViolation
+          | getRelevance a1 /= getRelevance b1 -> patternViolation neverUnblock -- Can we recover from this?
```

Comment:
No, I think this should actually be a proper error rather than a pattern violation.